crypto.go

```diff
@@ -22,6 +22,55 @@ const (
 // the output of a SHA256 hash.
 type Hash256 [sha256.Size]byte
 
+// SingleKeyECDH is an abstraction interface that hides the implementation of an
```

Comment:
Wondering still if `SingleKeyECDH` is the right name. It also suggests some fancy algorithm in which you exchange keys with yourself. Thinking about alternatives. Maybe `ExternalECDH` or just `Signer` (even though it isn't really signing)